packages/amplify-category-function/src/provider-utils/awscloudformation/utils/addLayerToFunctionUtils.ts

```diff
@@ -63,23 +64,38 @@ export const askLayerSelection = async (
   layerSelections = layerSelections.filter(selection => selection !== provideExistingARNsPrompt);
 
   for await (let layerName of layerSelections) {
-    const currentSelectionDefaults = filterProjectLayers(previousSelections).find(sel => sel.resourceName === layerName);
-    const currentVersion = currentSelectionDefaults ? currentSelectionDefaults.version.toString() : undefined;
     const layerVersions = await loadLayerDataFromCloud(context, layerName);
-    const layerVersionPrompt: ListQuestion = {
-      type: 'list',
-      name: 'versionSelection',
-      message: versionSelectionPrompt(layerName),
-      choices: layerVersions.map(layerVersion => layerVersion.Version.toString()),
-      default: currentVersion,
-    };
+    const layerVersionArrPrompt = layerVersions.map(layerVersion => layerVersion.Version.toString());
+    // skip asking version for a new layer
+    if (Array.isArray(layerVersionArrPrompt) && layerVersionArrPrompt.length) {
+      layerVersionArrPrompt.unshift(defaultLayerVersionPrompt);
+      const layerVersionPrompt: ListQuestion = {
+        type: 'list',
+        name: 'versionSelection',
+        message: versionSelectionPrompt(layerName),
+        choices: layerVersionArrPrompt,
+        default: defaultLayerVersionPrompt,
+      };
+      const versionSelection = (await inquirer.prompt(layerVersionPrompt)).versionSelection;
+      const isLatestVersionSelected = versionSelection.toString() === defaultLayerVersionPrompt ? true : false;
+      const selectedVersion = versionSelection.toString() === defaultLayerVersionPrompt ? layerVersionArrPrompt[1] : versionSelection;
+      lambdaLayers.push({
+        type: 'ProjectLayer',
+        resourceName: layerName,
+        version: selectedVersion,
+        isLatestVersionSelected: isLatestVersionSelected,
+        env: context.amplify.getEnvInfo().envName,
+      });
+    } else {
+      lambdaLayers.push({
+        type: 'ProjectLayer',
+        resourceName: layerName,
+        version: 1,
```

Comment:
We cannot have a hardcoded version, as mentioned in previous comment

Reply:
Here we are alraedy checking if the layer exists using ,loadLayerData from cloud. 
This info needs to stored in functionsParameter.json when we are deploying layer for first time. After first deployment , we populate this information using loadLayerData.

Reply:
can we set the version to undefined or null in that case?

Reply:
version is  set to default Prompt for new layers